packages/teams-js/src/public/video.ts

```diff
@@ -88,11 +88,33 @@ export namespace video {
     notifyError: (errorMessage: string) => void,
   ) => void;
 
+  /**
+   * Predefined failure reasons for preparing the selected video effect
+   * @beta
+   */
+  export enum EffectFailureReason {
+    /**
+     * A wrong effect id is provided
+     */
+    InvalidEffectId = 'InvalidEffectId',
+    /**
+     * The effect can't be initialezed
+     */
+    InitializationFailure = 'InitializationFailure',
+  }
+
+  enum EffectSuccessReason {
+    EffectPrepared = 'EffectPrepared',
+
+    NoPromiseReturned = 'NoPromiseReturned',
+  }
+
   /**
    * Video effect change call back function definition
+   * Return a Promise which will be resolved when the effect is prepared, or reject with a failure reason
```

Comment:
>reject with a failure reason [](http://example.com/codeflow?start=77&length=28)

Does this mean specifically that rejected promises should throw an `EffectFailureReason` value? You should explicitly document that and reference that structure.

Reply:
updated, please take a look